process/block/metablock.go

```diff
@@ -1107,6 +1110,34 @@ func (mp *metaProcessor) RevertStateToBlock(header data.HeaderHandler) error {
 	return nil
 }
 
+// RevertState recreates the state tries to the root hashes indicated by the provided header
+func (mp *metaProcessor) RevertState(currHeader data.HeaderHandler, prevHeader data.HeaderHandler) error {
+	err := mp.accounts.RecreateTrie(prevHeader.GetRootHash())
+	if err != nil {
+		log.Debug("recreate trie with error for header",
+			"nonce", prevHeader.GetNonce(),
+			"hash", prevHeader.GetRootHash(),
+		)
+
+		return err
+	}
+
+	err = mp.validatorStatisticsProcessor.RevertPeerState(prevHeader)
+	if err != nil {
+		log.Debug("revert peer state with error for header",
+			"nonce", prevHeader.GetNonce(),
+			"validators root hash", prevHeader.GetValidatorStatsRootHash(),
+		)
+
+		return err
+	}
+
+	mp.pruneStateAccountsOnRollback(currHeader, prevHeader)
```

Comment:
Might have extracted this in a separate function. In this way we will have only one function, not 2: RecreateStateTries and RevertState.

Reply:
Done.